data/database.py

```diff
@@ -1840,6 +1840,47 @@ class RepoMirrorConfig(BaseModel):
     root_rule = ForeignKeyField(RepoMirrorRule)
 
 
+@unique
+class IndexStatus(IntEnum):
+    """
+    Possible statuses of manifest security scan progress.
+    """
+
+    FAILED = -1
+    UNSUPPORTED = 0
+    TIMED_OUT = 1
+    WAITING = 2
+    IN_PROGRESS = 3
+    COMPLETED = 4
+
+
+@unique
+class IndexerVersion(IntEnum):
+    """
+    Possible versions of security indexers.
+    """
+
+    V2 = 0
+    V4 = 1
+
+
+class ManifestSecurityStatus(BaseModel):
```

Comment:
You should also add this to the transitive_deletion lists under User and Repository, to prevent recursive queries on delete